hyprtester/src/tests/main/window.cpp

```diff
@@ -1,22 +1,26 @@
-#include "tests.hpp"
-#include "../../shared.hpp"
-#include "../../hyprctlCompat.hpp"
-#include <print>
+#include <cmath>
 #include <thread>
 #include <chrono>
 #include <hyprutils/os/Process.hpp>
 #include <hyprutils/memory/WeakPtr.hpp>
-#include <csignal>
-#include <cerrno>
-#include "../shared.hpp"
 
-static int ret = 0;
+#include "../../shared.hpp"
+#include "../../hyprctlCompat.hpp"
+#include "../shared.hpp"
+#include "tests.hpp"
 
-using namespace Hyprutils::OS;
-using namespace Hyprutils::Memory;
+static int  ret = 0;
 
-#define UP CUniquePointer
-#define SP CSharedPointer
+static bool spawnKitty(const std::string& class_, int count) {
+    NLog::log("{}Spawning {}", Colors::YELLOW, class_);
+    if (!Tests::spawnKitty(class_)) {
+        NLog::log("{}Error: {} did not spawn", Colors::RED, class_);
+        return false;
+    }
+    NLog::log("{}Expecting {} window(s)", Colors::YELLOW, count);
+    EXPECT(Tests::windowCount(), count);
```

Comment:
Tests::spawnKitty already does this. The window count expect is redundant. You could just do EXPECT(Tests::spawnKitty(), true) iirc

the reason its there is because legacy. Window count tests can be dropped after kitty spawns